native/wasmex/src/instance.rs

```diff
@@ -225,6 +316,36 @@ pub enum WasmValue {
     F64(f64),
 }
 
+fn decode_term_as_wasm_value(expected_type: ValType, term: Term) -> Option<WasmValue> {
```

Comment:
I’ve copied and extract this from below. I could easily DRY this up by making the other use this function.

Reply:
I reused this.